### DISCUSSION TITLE
How should we suggest changes to the template?

### DISCUSSION THREAD

**Discussion by ryan-conway** [nimblehq/android-templates#417] [RFCs]
When updating the wiki, [we realised](https://github.com/nimblehq/android-templates/pull/409#discussion_r1114250596) there are 2 ways to suggest changes to the template:

1. Open an issue directly which only requires 3 votes
2. Open an RFC discussion which needs [50% of the vote](https://nimble-co.slack.com/archives/G41UCTGHL/p1676606571000989) (currently 7)

This would apply to new features and chores which require voting, with bugs still being created directly as issues.

Here's a quick table of pros and cons for each choice 🙏🏻 
| | Pros | cons |
| --- | --- | ---|
| Issue with 3 votes | - Current process so nothing needs to be changed | - 3 votes may not be enough do decide things _as a chapter_ <br/>- May not engage all members of the chapter<br/>- Can't make use of polls to simplify voting |
| RFC with 3 votes | - Can make use of polls <br/>- 3 votes mean features can be added quicker | - 3 votes may not be enough do decide things _as a chapter_ <br/>- May not engage all members of the chapter |
| RFC with 50% votes | - Can make use of polls <br/>- 50% of the vote means the majority of the chapter will be on the same page | - Too many votes may make the voting process longer |

**Comment by AVI5HEK** [comment]
Any pros and cons with each approach?

**Reply by ryan-conway** [reply]
@AVI5HEK Updated the RFC with a table of pros and cons 🙏🏻

**Comment by lydiasama** [comment]
I prefer both.

I preferably create any issues if it is an `improvement` or `bug` task. And it makes sense to get a `3-vote` indicator as it is approved to move on to another step.

And for discussing the new features or new approaches, i.e., updating coding style, libraries, etc., I think raising it as an RFC vote with a poll makes more sense. Because this action is required attention from all our chapter members. So getting more votes as much as possible can lighten us to more perspectives.

**Reply by ryan-conway** [reply]
Updating code style and libraries falls under chores tho [according to Compass](https://nimblehq.co/compass/product/backlog-management/user-stories/chores/#chore-definition) 😆  But I now agree that we should keep both methods for different purposes 😄 How does the following sound? 😇 

| Ticket type | Action | Required votes |
| --- | --- | --- |
| Feature | Create an RFC, follow with an issue if approved | 50% of the team, currently 7 |
| Chore | Create an issue directly with `need voting` label | 3 |
| Bug | Create an issue directly | 0 since bugs are considered essential |

**Reply by toby-thanathip** [reply]
@lydiasama @ryan-conway Your suggestion looks good, but would you agree it feels a bit too overcomplicated? 💭 

IMO, I think we should simplify by creating an RFC for both `features` & `chores`.

Because even `chores` (Ex: Updating Gradle) will require extra discussion at times, some other examples:

- https://github.com/nimblehq/android-templates/discussions/257
- https://github.com/nimblehq/android-templates/discussions/396

Voted ✅

**Reply by chornerman** [reply]
Agree with @Tuubz 🚀

**Comment by toby-thanathip** [comment]
@ryan-conway If you agree with the final result, we can resolve this discussion 🚀
